platform/broadcom/sonic-platform-modules-dell/s5296f/sonic_platform/component.py

```diff
@@ -0,0 +1,233 @@
+    return ''
+
+def get_cpld0_version():
+    return get_cpld_version(601, 0x31)
+
+def get_cpld1_version():
+    return get_cpld_version(600, 0x30)
+
+def get_cpld2_version():
+    return get_cpld_version(600, 0x31)
+
+def get_cpld3_version():
+    return get_cpld_version(600, 0x32)
+
+def get_cpld4_version():
+    return get_cpld_version(600, 0x33)
+
+
+class Component(ComponentBase):
+    """DellEMC Platform-specific Component class"""
+
+    CHASSIS_COMPONENTS = [
+        ['BIOS',
+         'Performs initialization of hardware components during booting',
+         get_bios_version
+         ],
+
+        ['FPGA',
+         'Used for managing the system LEDs',
+         get_fpga_version
+         ],
+
+        ['BMC',
+         'Platform management controller for on-board temperature monitoring, in-chassis power, Fan and LED control',
+         get_bmc_version
+         ],
+
+        ['System CPLD',
+         'Used for managing the CPU power sequence and CPU states',
+         get_cpld0_version
+         ],
+
+        ['Secondary CPLD 1',
+         'Used for managing SFP28/QSFP28 port transceivers (SFP28 1-24, QSFP28 1-4)',
+         get_cpld1_version
+         ],
+
+        ['Secondary CPLD 2',
+         'Used for managing SFP28/QSFP28 port transceivers (SFP28 25-48, QSFP28 5-8)',
+         get_cpld2_version
+         ],
+
+        ['Secondary CPLD 3',
+         'Used for managing SFP28/QSFP28 port transceivers (SFP28 49-72)',
+         get_cpld3_version
+         ],
+
+        ['Secondary CPLD 4',
+         'Used for managing SFP28/QSFP28 port transceivers (SFP28 73-96)',
+         get_cpld4_version
+         ]
+    ]
+
+    def __init__(self, component_index = 0):
+        self.index = component_index
+        self.name = self.CHASSIS_COMPONENTS[self.index][0]
+        self.description = self.CHASSIS_COMPONENTS[self.index][1]
+        self.version = self.CHASSIS_COMPONENTS[self.index][2]()
+
+    def get_name(self):
+        """
+        Retrieves the name of the component
+        Returns:
+        A string containing the name of the component
+        """
+        return self.name
+
+    def get_description(self):
+        """
+        Retrieves the description of the component
+        Returns:
+        A string containing the description of the component
+        """
+        return self.description
+
+    def get_firmware_version(self):
+        """
+        Retrieves the firmware version of the component
+        Returns:
+            A string containing the firmware version of the component
+        """
+        return self.version
+
+    def get_presence(self):
+        """
+        Retrieves the presence of the component
+        Returns:
+            bool: True if  present, False if not
+        """
+        return True
+
+    def get_model(self):
+        """
+        Retrieves the part number of the component
+        Returns:
+            string: Part number of component
+        """
+        return 'NA'
```

Comment:
will this be added later?

Reply:
Components described here are software application like BIOS,CPLD,FPGA,.. for which we don't have model or serial number.